lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -690,14 +690,16 @@ function dosomething_reportback_save($values) {
   }
   else {
     $entity = entity_load_single('reportback', $rbid);
+    $reportback_item = entity_load_single('reportback_item', $values['fid']);
   }
   try {
     // Set entity properties.
     dosomething_reportback_set_properties($entity, $values);
     // Set entity files.
-    dosomething_reportback_set_files($entity, $values);
-    // Save the entity.
+    dosomething_reportback_set_files($reportback_item, $values);
```

Comment:
@angaither when it was saved to the reportback entity, I wasn't able to access `$entity` after `$entity->save();` and it seemed to break the code, never reaching the next `if` statement and then redirecting incorrectly since it didn't have the `$entity->rbid` to return back to lines 359-366 in `dosomething_preortback_form_submit` ( `dosomething_reportback.forms.inc` file). 

I'm not sure what you mean by set the files on the reportback entity and then also save the reportback_item entity info?